<html>
    <head>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
        <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
        <link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='CSS/base.css') }}" />
        <script src="{{ url_for('static', filename='JS/base.js') }}"></script>
    </head>
    <body>  
        <title>Metagenomics Visualize Platform</title>
        <h1>Metagenomics Visualize Platform</h1>
        <h2>Upload File</h2>
        <div id='div2'>
            1. upload your File.<br>
            2.select which item you are interested in .
            <br>
            <br>
    <form method=post enctype=multipart/form-data>
      <input type=file name=file id='upload'>
      <input type=submit value=Upload >
    </form>
    </div>
    <h2>Annotation</h2>

    <section>
        <div id='tree_result'>TREE PLOT</div>
        <div id='anno_result'>anno result will be plotted here.</div>
    </section>
    <section>
        <div id='heatmap_result'>heatmap</div>
        <div id='alpha_diversity_result'>alpha diver</div>
        <div id='beta_diversity_result'>beta diver</div>
    </section>
    <br>
    we want to assert something to here.
    <br>
    <section>
       <div id='plot_subtree' ">subtree</div>
       <div id='sub_tree_anno'>subtree anno</div>
       <div id='whole_abu_GI_result'>whole treee abu and gi</div>
    </section>
    <br>
    maybe the next web
    <br>
    <section>
       <div id='corr_abu_result'>corr_abu</div>
       <div id='corr_GI_result'>corr_GI</div>
    </section>
    <br>

    <section>
        <div id='stat_test_result'>stat_test_result</div>
        <div id='osea_result'>osea_result</div>
    </section> 
    <br>

    <div>
        Taxon:<input id='taxonomy_file' value='taxonomy.tsv'>
        <br>
        Feature_table:<input id='feature_table_file' value='feature-table.biom'>
        <br>
        <br>
    </div>
            
        <h2>Tree</h2>
        <div>
            <p>you can upload sequences to generate a phylogenetic tree,or upload newick tree to 
                draw a tree
            </p>
        </div>
        <div id='div3'>
            tree_file_name:<input id='tree_file_name' value='tree.nwk'>
            tree_format:<select id='tree_format'>
                <option>newick</option>
                <option>phyloxml</option>
            </select>
            <br><br>

        </div>
        <br>
        input the node num of the subtree root:<input id='node_num' value=0>
        <select id='tree_type'>
            <option>rectangle_tree</option>
            <option>circular_tree</option>
        </select>
        <button id= 'plot_tree' onclick="plot_tree()">plot tree</button>
        <div>
            <h2>Matrix</h2>

        feature table file name :<input id='feature_table' value='feature-table.biom'>(e.g.: "experiment1.otutable" without quote)
        </div>
        <br>
        <div>
             sample metadata file name:<input id='metadata' value='demo_metadata.tsv'>
        </div>
        <br>
        <br>
        <h3>filt matrix</h3>
        Select number of features to be analysis default:<input id='Num_features' value=1>
        <button id='feature_num' onclick="copy_options()">feature_ num_confirm</button>
        <br>
        Prevalence:<input id='prevalence' value=0.1>
        <br>
        Abundance:<input id='abundance' value=100>
        <br>
        Variance:<input id='variance' value=100>
        <br>
        <script>
            function copy_options(){
            var num_features =document.getElementById("Num_features").value;
            if(num_features > 1){
                $('#mySelect0 option').clone().appendTo('#mySelect1')
                if(num_features > 2){
                $('#mySelect0 option').clone().appendTo('#mySelect2')
                }
            }
            }
        </script> 
        <br>
         <button id='read_metadata'onclick="read_metadata()">read metadata</button> 
        <br>
        <h2>Abundance</h2>
        <div>abundance type:
        <select id='abundance_type'>
            <option>absolute</option>
            <option>realtive</option>
        </select>
             </div>
        <br>
        <div>obtain log abundance: 
        <select id='log_flag'>
            <option>yes</option>
            <option>no </option>
        </select>
        </div>
        <button id='plot_abun_result' onclick="plot_abun()">plot abundance</button>
        <div id='abun_result'style="width:800px;height:600px;"></div>
        Feature to be analysied:
            <select id="mySelect0" size=1>
                <option>None</option>
            </select>
            <select id="mySelect1" size=1 type="hidden">
            </select>
            <select id="mySelect2" size=1 type="hidden">
            </select>
          <br>
        <button id="plot_heatmap" onclick="send_heatmap()">plot heatmap</button>
        <br>
        <br>
        <br>
        <br>
        <br>
        <h2>statistical methods</h2>
        <div id='stats description'>You can do some test and correlation 
            compute here.For example, t-test or F-test between two groups
        labled 0 and 1.</div>
        <form>
            <select id="statistics methods">
                <option>t_test</option>
                <option>F_test</option>
                <option>some-other-test-methods</option>
                <option>TO BE CONTINUED..</option>
            </select>
        </form>
        statistical result:
        <button id='gen_result' onclick="plot_stats_result()">generate result</button>
        




        <h2>OTU Set Enrichment Analysis(OSEA)</h2>
        choose the rank list.<br>
        choose the lineage level(e.g. phylum ,genus,specied.etc)
        <select id='set_level'>
            <option>Kingdom</option>
            <option>Phylum</option>
            <option>Class</option>
            <option>Order</option>
            <option>Family</option>
            <option>Genus</option>
            <option>Species</option>
        </select>
        <button id='gen_osea_result' onclick="plot_osea_result()">generate result</button>
        
       <h2>Dimension Reduction</h2>
       n_component(which deicde the number of dimensions after dimension reduction):
       <select id='flag_3d'>
           <option>False</option>
           <option>True</option>
       </select> 
       <br>
       <select id='n_component'>
           <option>2</option>
           <option>3</option>
       </select> 
       <br>
       <select id='method'>
           <option>Isomap</option>
           <option>MDS</option>
       </select> 
       <button id='gen_dim_reduce_result' onclick="plot_dim_reduce_result()">generate dim_reduce result</button>
       <div id='dim_reduce_result' style="width:800px;height:600px">dim_reduce_result</div>

       <h2>Diversity</h2>
       <h3>alpha diversity</h3>
        alpha table file name :<input id='alpha_table_file' value='alpha-diversity.tsv'>
       <button id='gen_alpha_diversity_result' onclick="plot_alpha_div()">plot alpha div</button>
       <select id='alpha_metric'>
       <option>faith_pd</option>
       <option>ace</option>
       <option>chao1</option>
       <option>fisher_alpha</option>
       <option>gini_index</option>
       <option>goods_coverage</option>
       <option>robbins</option>
       <option>shannon</option>
       <option>simpson</option>
       <option>simpson_e</option>
       </select>
        distance matrix file name :<input id='distance_matrix_file' value='distance-matrix.tsv'>
        <h2>rarefaction</h2>
        <button id='gen_rarefaction' onclick="plot_alpha_rarefaction()">plot rarefaction</button> 
        <select id='rarefaction_metric'>
            <option>faith_pd</option>
            <option>ace</option>
            <option>chao1</option>
            <option>fisher_alpha</option>
            <option>gini_index</option>
            <option>goods_coverage</option>
            <option>robbins</option>
            <option>shannon</option>
            <option>simpson</option>
            <option>simpson_e</option>
        </select>
        max_sequence depth:<input id='max_seq' value=300>
        step:<input id ='step' value=50>
        rarefied number <input id='rarefied_num' value=10>
        <div id='rarefaction_box_result' style="width:800px;height: 600px"></div>
        <div id='rarefaction_scatter_result' style="width:800px;height: 600px"></div>
       
       <h3>beta diversity</h3>

       <select id='beta_metric'>
           <option>weighted_unifrac</option>
           <option>unweighted_unifrac</option>
           <option>Bray_Curtis</option>
       </select>

       <select id='beta_dim_method'>
           <option>PCoA</option>
           <option>Isomap</option>
           <option>MDS</option>
       </select>

       <select id='beta_n_components'>
           <option>2</option>
           <option>3</option>
       </select>

       <button id='gen_beta_diversity_result' onclick="plot_beta_div()">plot beta div</button>
       <h2>ecology analysis</h2>
       <h3>phylogenetic tree</h3>
        node-number :<input id='ecology_tree_node_num' value=0>
       <button id='phylo_tree_generator' onclick="plot_phylo_tree()">plot phylo tree</button>
       <div id='phylo_tree_result' style="width:800px;height: 600px"></div>

       <h3>tree and scatters</h3>
       For categorical metadata(like "yes" or "no" in some column),we prefer use the statistics test methods.
       statistical test methods:
       <select id = 'stats_test'>
           <option>None</option>
           <option>t_test</option>
           <option>F_test</option>
           <option>mannwhitneyu</option>
       </select>
       <br>
       For numerical metadata(like Age),we prefer use the correlation Analysis.
       correlation coefficient methods:
       <select id="corr_coef">
           <option>None</option>
           <option>spearman</option>
           <option>pearson</option>
       </select>
       <br>

       subtree root node ID number <input id='ID_num' value=0>
       <button id='ecology_scatter_generator' onclick="plot_ecology_scatter()">plot ecology_scatter</button>
    </body>

</html>